numpy/core/tests/test_scalar_methods.py

```diff
@@ -0,0 +1,79 @@
+"""
+Test the scalar constructors, which also do type-coercion
+"""
+from __future__ import division, absolute_import, print_function
+
+import fractions
+import numpy as np
+
+from numpy.testing import (
+    run_module_suite,
+    assert_equal, assert_almost_equal, assert_raises, assert_warns,
+    dec
+)
+
+float_types = [np.half, np.single, np.double, np.longdouble]
+
+def test_float_as_integer_ratio():
+    # derived from the cpython test "test_floatasratio"
+    for ftype in float_types:
```

Comment:
Could now use `@pytest.mark.parametrize` but no big deal.

Reply:
Working on this locally a bit, I ended up splitting into a few different tests, with a really small one that looks like this to cut down on the nesting:

```python
 16 @pytest.mark.parametrize("ftype", [
 17     np.half, np.single, np.double, np.longdouble])
 18 @pytest.mark.parametrize("f, ratio", [
 19     (0.875, (7, 8)),
 20     (-0.875, (-7, 8)),
 21     (0.0, (0, 1)),
 22     (11.5, (23, 2)),
 23     ])
 24 def test_float_as_integer_ratio(ftype, f, ratio):
 25     # derived from the cpython test "test_floatasratio"
 26     assert_equal(ftype(f).as_integer_ratio(), ratio)
```

Reply:
this have been parametrized now, so marking as resolved